src/plugins/chart_expressions/expression_xy/public/components/xy_chart.tsx

```diff
@@ -157,50 +143,44 @@ export function XYChart({
   const chartTheme = chartsThemeService.useChartsTheme();
   const chartBaseTheme = chartsThemeService.useChartsBaseTheme();
   const darkMode = chartsThemeService.useDarkMode();
-  const filteredLayers = getFilteredLayers(layers, data);
-  const layersById = filteredLayers.reduce<Record<string, XYLayerConfigResult>>(
-    (hashMap, layer) => {
-      hashMap[layer.layerId] = layer;
-      return hashMap;
-    },
+  const filteredLayers = getFilteredLayers(layers);
+  const layersById = filteredLayers.reduce<Record<string, CommonXYLayerConfig>>(
+    (hashMap, layer) => ({ ...hashMap, [layer.layerId]: layer }),
     {}
   );
 
   const handleCursorUpdate = useActiveCursor(chartsActiveCursorService, chartRef, {
-    datatables: Object.values(data.tables),
+    datatables: filteredLayers.map(({ table }) => table),
   });
 
   if (filteredLayers.length === 0) {
-    const icon: IconType = getIconForSeriesType(getDataLayers(layers)?.[0]?.seriesType || 'bar');
+    const icon: IconType = getIconForSeriesType(
+      getDataLayers(layers)?.[0]?.seriesType || SeriesTypes.BAR
+    );
     return <EmptyPlaceholder className="xyChart__empty" icon={icon} />;
   }
 
+  const dataLayers: CommonXYDataLayerConfig[] = filteredLayers.filter(isDataLayer);
+
   // use formatting hint of first x axis column to format ticks
-  const xAxisColumn = data.tables[filteredLayers[0].layerId].columns.find(
-    ({ id }) => id === filteredLayers[0].xAccessor
-  );
+  const xAxisColumn = dataLayers[0]?.table.columns.find(({ id }) => id === dataLayers[0].xAccessor);
 
   const xAxisFormatter = formatFactory(xAxisColumn && xAxisColumn.meta?.params);
-  const layersAlreadyFormatted: Record<string, boolean> = {};
+  const areLayersAlreadyFormatted = getAreAlreadyFormattedLayersInfo(dataLayers, formatFactory);
 
   // This is a safe formatter for the xAccessor that abstracts the knowledge of already formatted layers
   const safeXAccessorLabelRenderer = (value: unknown): string =>
-    xAxisColumn && layersAlreadyFormatted[xAxisColumn.id]
+    xAxisColumn && areLayersAlreadyFormatted[dataLayers[0]?.layerId]?.[xAxisColumn.id]
       ? String(value)
       : String(xAxisFormatter.convert(value));
 
   const chartHasMoreThanOneSeries =
     filteredLayers.length > 1 ||
     filteredLayers.some((layer) => layer.accessors.length > 1) ||
-    filteredLayers.some((layer) => layer.splitAccessor);
-  const shouldRotate = isHorizontalChart(filteredLayers);
-
-  const yAxesConfiguration = getAxesConfiguration(
-    filteredLayers,
-    shouldRotate,
-    data.tables,
-    formatFactory
-  );
+    filteredLayers.some((layer) => isDataLayer(layer) && layer.splitAccessor);
+  const shouldRotate = isHorizontalChart(dataLayers);
+
+  const yAxesConfiguration = getAxesConfiguration(filteredLayers, shouldRotate, formatFactory);
```

Comment:
Reference line layers should not be taken into consideration for the axis assignment, only data layers

Reply:
Thanks. Makes sense. Didn't want to change the logic of the existing implementation not to brake anything.

Reply:
Done.